libraries/security/src/main/java/org/dataportabilityproject/security/RsaSymmetricKeyGenerator.java

```diff
@@ -39,4 +43,21 @@ public KeyPair generate() {
     kpg.initialize(1024);
     return kpg.genKeyPair();
   }
+
+  @Override
+  public PublicKey parse(byte[] encoded) {
+    KeyFactory factory;
+    try {
+      factory = KeyFactory.getInstance(ALGORITHM);
+    } catch (NoSuchAlgorithmException e) {
+      logger.error("NoSuchAlgorithmException for: {}", ALGORITHM, e);
+      throw new RuntimeException("NoSuchAlgorithmException generating key", e);
+    }
+    try {
+      return factory.generatePublic(new X509EncodedKeySpec(encoded));
+    } catch (InvalidKeySpecException e) {
+      logger.error("InvalidKeySpecException for: {}", ALGORITHM, e);
+      throw new RuntimeException("InvalidKeySpecException generating key", e);
+    }
```

Comment:
can we combine these try catches? (is that ok stylewise?) Since they both just log and rethrow,  it might be cleaner to combine

Reply:
Done